master/internal/sproto/task.go

```diff
@@ -17,10 +17,13 @@ type (
 		// Identifying information.
 		AllocationID      model.AllocationID
 		TaskID            model.TaskID
-		JobID             *model.JobID
-		JobSubmissionTime *time.Time
-		State             job.SchedulingState
-		Name              string
+		JobID             model.JobID
+		JobSubmissionTime time.Time
+		// IsUserVisible determines whether the AllocateRequest should
+		// be considered in user-visible reports.
+		IsUserVisible bool
+		State         job.SchedulingState
```

Comment:
note: i didn't notice until now this state was added. why is it on the allocation request, shouldn't the request be immutable basically? looking through the code, this seems wrong. i think taskList needs some repr of it's side of things rather than updating this request object. also, to not it there is a lot of overlap between job.SchedulingState and model.AllocationState and we should figure out where we're going with that

Reply:
there's probably a better way of tracking that. we look at the resource assignment info when an assignment happens to set this. (minus backfilling)  we could have tracked this separately on a job_list struct or by looking up the assignment on each read. I'd need to think if we can track backfill status with the latter.

Reply:
> also, to not it there is a lot of overlap between job.SchedulingState and model.AllocationState and we should figure out where we're going with that.

👍🏼  I'd definitely chat more before making changes to the schedulingstate to see what you think. I see scheduling as a higher level state but helps to see where they overlap and where they don't